File: Java/3.Arrays/sorting/BubbleSort.java
public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = {5, 8, 4, 1, 2, 3, 6, 9, 7};
        int n = arr.length;

        System.out.println("Bubble Sort: ");

        for (int i = 0; i < n-1; i++) { // total round
            for (int j = 0; j < n-i-1; j++) { 
                if (arr[j] > arr[j+1]) {
                    //swap arr[j] & arr[j+1]
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }

        //print the sorted array
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}

File: C++/4.Arrays & String/3.Sorting/bubbleSort.cpp
#include <iostream>
using namespace std;

void swap (int *xp, int *yp) {
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

void bubbleSort (int a[], int n) {

    for (int i = 0; i < n-1; i++) {
        bool swapped = false;
        for (int j = 0; j < n-i-1; j++) {
            if (a[j] > a[j+1]) {
                swap (&a[j], &a[j+1]);
                swapped = true;
            }
        }
        if (swapped == false)
            break;
    }   
}

//print sorted array element
void printArray(int a[], int n) {
    for (int i = 0; i < n; i++) {
        cout << a[i] << " ";
    } cout << endl;    
}

int main() {
    
    int arr[] = {5, 4, 2, 6, 3, 9, 7};

    int size = sizeof(arr) / sizeof (arr[0]);

    cout <<"Bubble Sort: "<<endl;
    bubbleSort (arr, size);
    printArray(arr, size);

    return 0;
}
